adapters/goal.ts

```diff
@@ -1,49 +1,60 @@
-import axios from 'axios'
-import * as goalAdapter from './goal.transformer'
-import { type Goal } from '~/interfaces/Goal'
-import { type PostGoal } from '~/interfaces/PostGoal'
-import { API } from '~/constants/api'
+import axios from "axios";
+import * as goalAdapter from "./goal.transformer";
+import { type Goal } from "~/interfaces/Goal";
+import { type PostGoal } from "~/interfaces/PostGoal";
+import { API } from "../constants/api";
 
 const goalSiteConfig = {
   headers: {
-    'Content-Type': 'application/vnd.api+json'
-  }
-}
+    "Content-Type": "application/vnd.api+json",
+  },
+};
 
-export const fetchGoals = async (): Promise<Goal []> => {
+export const fetchGoals = async (): Promise<Goal[]> => {
   const goals: Goal[] = await axios
     .get(`${API.GOAL_BASE_URL}/goal/`)
-    .then(res => goalAdapter.transformGoalsFromApi(res.data.data))
-    .catch(error => {
-      throw new Error(error)
-    })
+    .then((res) => goalAdapter.transformGoalsFromApi(res.data.data))
+    .catch((error) => {
+      throw new Error(error);
+    });
 
-  return goals
-}
+  return goals;
+};
 
 export const addGoal = async (goal: PostGoal): Promise<Goal> => {
   const goalResponse = await axios
-    .post(`${API.GOAL_BASE_URL}/goal/`, {
-      data: {
-        type: 'Goal',
-        attributes: goal
-      }
-    }, goalSiteConfig)
-    .then(res => goalAdapter.transformGoalFromApi(res.data.data))
+    .post(
+      `${API.GOAL_BASE_URL}/goal/`,
+      {
+        data: {
+          type: "Goal",
+          attributes: goal,
+        },
+      },
+      goalSiteConfig
+    )
+    .then((res) => goalAdapter.transformGoalFromApi(res.data.data));
 
-  return goalResponse
-}
+  return goalResponse;
+};
 
-export const updateGoal = async (goalId: string, goal: PostGoal): Promise<Goal> => {
+export const updateGoal = async (
```

Comment:
here the function  is not checking if the goalId parameter is a valid ID. This means that if the goalId parameter is not a valid ID, the axios request will fail, and the updateGoal() function will not return a Goal object.